doc/ref_arch/openstack/chapters/chapter03.md

```diff
@@ -185,7 +185,8 @@ This section describes the core set of services and service components needed to
 
 <a name="3.3.2"></a>
 ### 3.3.2. Tenant Isolation
-Content to be developed
+In OpenStack user interfaces and documentation, a group of users is referred to as a project or tenant. With Keystone v1 and v2(both deprecated), the term "tenant" was used. With Keystone v3, the term "project" got adopted and both the terms became interchangeable. However, as CNTT RA uses Keystone v3 in [this](chapter05.md#5.2) section, so it is recommended to use the term "project" when referring to OpenStack and use [tenant](../../ref_model/chapters/chapter03.md#321-tenant) when referring to multi-tenancy . According to [OpenStack glossary](https://docs.openstack.org/doc-contrib-guide/common/glossary.html), Projects represent the base unit of "ownership" in OpenStack, in that all resources in OpenStack should be owned by a specific project.
+OpenStack offers multi-tenancy by means of resource(compute, network and storage)separation via projects. OpenStack offers ways to share virtual resources between tenants while maintaining logical separation. As an example, traffic separation is provided by creating different VLAN ids for different tenant neutron networks. As another example, if host separation is needed, nova scheduler offers AggregateMultiTenancyIsolation scheduler filter to separate tenants in host aggregates. Thus, if a host in an aggregate is configured for a particular tenant, only the instances from that tenant are placed on the host. Overall, tenant isolation ensures that the resources of a project are not affected by resources of another project.
```

Comment:
Can the term "project" be used consistently -- replace "tenant" with "project" except in phrases such as "tenant isolation."

Reply:
Fixed, thanks !